packages/core/src/use-params.ts

```diff
@@ -47,40 +48,49 @@ export function useParams(returnType: "array"): Record<string, Array<string | un
 export function useParams(returnType?: "string" | "number" | "array") {
   const router = useRouter()
   const query = useRouterQuery()
+  const [lastRouterQuery, setLastRouterQuery] = useState(router.query)
 
-  const rawParams = extractRouterParams(router.query, query)
+  if (!isEqual(router.query, lastRouterQuery)) {
+    setLastRouterQuery(router.query)
+  }
```

Comment:
1. We can't setState in the render method because render methods must be pure (especially for concurrent mode)
2. Can we just use `router.query` directly instead of `lastRouterQuery`?